src/ansi-c/library/pthread_lib.c

```diff
@@ -290,12 +290,15 @@ inline int pthread_mutex_destroy(pthread_mutex_t *mutex)
 #define __CPROVER_PTHREAD_H_INCLUDED
 #endif
 
-extern __CPROVER_bool __CPROVER_threads_exited[];
-extern __CPROVER_thread_local unsigned long __CPROVER_thread_id;
-
-extern __CPROVER_thread_local const void *__CPROVER_thread_keys[];
-extern __CPROVER_thread_local void (*__CPROVER_thread_key_dtors[])(void *);
-extern __CPROVER_thread_local unsigned long __CPROVER_next_thread_key;
+__CPROVER_bool __CPROVER_threads_exited[__CPROVER_constant_infinity_uint];
+__CPROVER_thread_local unsigned long __CPROVER_thread_id = 0;
+#if 0
```

Comment:
Any particular reason for the `#if 0` rather than deletion? If theres a reason (other than "its a work in progress" :-) ) I'd suggest adding a comment to explain the reason.

Reply:
Comments added in all the places that have `#if 0`: thread key destructors are expensive. We'll have to figure out a viable solution at some point.

Reply:
Thanks, much appreciated :-) I always hate encountering an `#if 0` with not context as to why its there, whether it works if re-enabled, etc.